tests/test_component.py

```diff
@@ -0,0 +1,94 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for the microgrid component wrapper."""
+
+import frequenz.api.common.components_pb2 as components_pb
+import frequenz.sdk.microgrid.component._component as cp
+import pytest
+
+# pylint:disable=no-member
+
+
+# pylint: disable=protected-access
+def test_component_category_from_protobuf() -> None:
+    """Test the creating component category from protobuf."""
+    assert (
+        cp._component_category_from_protobuf(
+            components_pb.ComponentCategory.COMPONENT_CATEGORY_UNSPECIFIED
+        )
+        == cp.ComponentCategory.NONE
+    )
+
+    assert (
+        cp._component_category_from_protobuf(
+            components_pb.ComponentCategory.COMPONENT_CATEGORY_GRID
+        )
+        == cp.ComponentCategory.GRID
+    )
+
+    assert (
+        cp._component_category_from_protobuf(
+            components_pb.ComponentCategory.COMPONENT_CATEGORY_METER
+        )
+        == cp.ComponentCategory.METER
+    )
+
+    assert (
+        cp._component_category_from_protobuf(
+            components_pb.ComponentCategory.COMPONENT_CATEGORY_INVERTER
+        )
+        == cp.ComponentCategory.INVERTER
+    )
+
+    assert (
+        cp._component_category_from_protobuf(
+            components_pb.ComponentCategory.COMPONENT_CATEGORY_BATTERY
+        )
+        == cp.ComponentCategory.BATTERY
+    )
+
+    assert (
+        cp._component_category_from_protobuf(
+            components_pb.ComponentCategory.COMPONENT_CATEGORY_EV_CHARGER
+        )
+        == cp.ComponentCategory.EV_CHARGER
+    )
+
+    assert cp._component_category_from_protobuf(666) == cp.ComponentCategory.NONE  # type: ignore
```

Comment:
Interesting.
`666` I assume is just a number that is not present in the `components_pb.ComponentCategory` enum? If yes, then getting a value like this would mean that something has gone horribly wrong, e.g., an API version mismatch.
Do we still want to return a `None` in this case, instead of an error?

Reply:
Good question. In any case IMHO this is for a separate issue, this PR is just importing the code we have in the SDK right now. I'm not sure if @shsms wrote this originally and if he could remember if there was a particular reason for this.

Reply:
Fair enough. I'll suggest creating an issue to track this.

Reply:
OK, I will create discussions for the topics that are not clear we want to change, and issues for the ones that are obviously wrong and need fixing.

* https://github.com/frequenz-floss/frequenz-client-microgrid-python/discussions/14

Reply:
I didn't write these, these are from before I started working on the SDK.